src/librustc_ast/token.rs

```diff
@@ -448,13 +448,22 @@ impl Token {
     /// Returns `true` if the token is any literal, a minus (which can prefix a literal,
     /// for example a '-42', or one of the boolean idents).
     ///
-    /// Keep this in sync with `Lit::from_token`.
+    /// In other words, would this token be a valid start of `parse_literal_maybe_minus`?
+    ///
+    /// Keep this in sync with and `Lit::from_token`, excluding unary negation.
     pub fn can_begin_literal_or_bool(&self) -> bool {
```

Comment:
```suggestion
    pub fn can_begin_literal_maybe_minus(&self) -> bool {
```
This should probably be renamed to avoid surprises.

Reply:
Done.